### DISCUSSION TITLE
completion of slash commands

### DISCUSSION THREAD

**Discussion by Konfekt** [olimorris/codecompanion.nvim#211] [Ideas]
Thank you for this plug-in! How to complete slash commands (in the sense of `:help command-completion-custom`) on the command line by the prompts listed in the dictionary at https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua  ?

**Comment by S1M0N38** [comment]
At the moment sub-command completion using slash-command is not supported in command mode,\
 i.e `:CodeCompanion /slash-cmd`.

However in chat buffer auto-completion is enabled (see README)

**EDIT:**
Btw, recently I've implement something really similar https://github.com/S1M0N38/dante.nvim/blob/main/plugin/dante.lua
A good reference for implementing autocompletion in sub-commands is [nvim-best-practices](https://github.com/nvim-neorocks/nvim-best-practices?tab=readme-ov-file#speaking_head-user-commands)

**Reply by Konfekt** [reply]
> A good reference for implementing autocompletion in sub-commands is [nvim-best-practices](https://github.com/nvim-neorocks/nvim-best-practices?tab=readme-ov-file#speaking_head-user-commands)

This reads more complicated than `:help command-completion-custon`
